src/domain/accounts/index.ts

```diff
@@ -22,11 +22,22 @@ export * from "./primitives"
 const UserIdRegex =
   /^[0-9a-f]{8}-[0-9a-f]{4}-[1-5][0-9a-f]{3}-[89ab][0-9a-f]{3}-[0-9a-f]{12}$/i
 
+// device id format from AppCheck: 1:72279297366:android:35666807ae916c5aa75af7
+const DeviceIdRegex = /^[0-9]+:[0-9]+:[a-z]+:[0-9a-z]+$/i
+
 export const checkedToUserId = (userId: string): UserId | ValidationError => {
-  if (!userId.match(UserIdRegex)) {
-    return new InvalidUserId(userId)
+  // kratosUserId is a uuid v4
+  if (userId.match(UserIdRegex)) {
```

Comment:
Here it becomes very obvious our domain model is insufficient.

Imo it should be something like this:
```
type AuthSubjectRef = AppCheckDeviceId | KratosId
```
and `UserId` is a mongoldb database reference that is immutable.

Reply:
I defer to @nicolasburtey . maybe we can do a data modeling session in a follow up PR. Below nicolas created a diagram of the existing model.  This shows the Account collection as the source of truth (top of the heirarchy). Our current data model has some duplication and can probably be normalized in a follow up PR. i.e user id is duplicated in the user collection as the foreign key link, but in a normalized database should probably be AccountId 
** Entity represents the kratos database
<img width="632" alt="Screen Shot 2023-06-01 at 4 56 58 PM" src="https://github.com/GaloyMoney/galoy/assets/1273575/c40b5ea2-5176-46bf-941f-948240cb83a8">